docs/Developers/Debug_Bears.rst

```diff
@@ -0,0 +1,77 @@
+Debug Bears
+===========
+
+This document provide a brief overview of coala's Debugging Interface.
+The Debugging Interface will help users to debug Bear code and step through
+it using pdb interface
+
+Usage of coala's Debugger
+=========================
+
+Command Line Interface
+----------------------
+
+Users can specify the bear they want to debug using ``--debug-bears``, i.e.
+
+.. code:: shell
+
+    $coala -b PEP8Bear,HelloWorldBear -f <filename> --debug-bears HelloWorldBear
+
+Below is the code given for a simple ``HelloWorldBear`` that prints a debug
+message for each file:
+
+.. code:: python
+
+    import logging
+
+    from coalib.bears.LocalBear import LocalBear
+
+
+    class HelloWorldBear(LocalBear):
+        def run(self,
+                filename,
+                file):
+            logging.info('Please follow newcomer guide.')
+            logging.debug('Hello World! Checking file {}.'.format(filename))
+
+After invoking the coala's debugger it will step into the ``run()`` method of a
+HelloWorldBear and help users to debug Bear code with passed settings.
```

Comment:
Can you try to move the explanation part (describing how debugging exactly works) above "Usage" paragraph? In some different paragraph like "Details" or "Behaviour" (or any name you like that shows that we want to describe what the idea is and how things work :) ).
So the usage part in the end only concentrates on invoking the debugger.

That would make texts more consistent across different usage patterns (CLI vs coafile).